test/Record.purs

```diff
@@ -196,6 +198,69 @@ test =
     , c: 42
     }
 
+-----
+-- Verify that multiple maps can be used in constraints
+
+newtype ReplaceLeft r = ReplaceLeft { | r }
+
+instance replaceLeftH ::
+  (IsSymbol sym, Row.Cons sym a x vals) =>
+  MappingWithIndex (ReplaceLeft vals) (SProxy sym) (Either a b) (Either a b) where
+  mappingWithIndex (ReplaceLeft vals) prop = lmap (const $ Record.get prop vals)
+
+replaceLeft :: forall rvals rin rout.
+  HMapWithIndex (ReplaceLeft rvals) { | rin } { | rout } =>
+  { | rvals } ->
+  { | rin  } ->
+  { | rout }
+replaceLeft =
+  hmapWithIndex <<< ReplaceLeft
+
+testReplaceLeft :: _
+testReplaceLeft =
+  { a: "goodbye"
+  , b: 100
+  }
+  `replaceLeft`
+  { a: Left "hello"
+  , b: Right 1
+  }
+
+newtype ReplaceRight r = ReplaceRight { | r }
+
+instance replaceRightH ::
+  (IsSymbol sym, Row.Cons sym b x vals) =>
+  MappingWithIndex (ReplaceRight vals) (SProxy sym) (Either a b) (Either a b) where
+  mappingWithIndex (ReplaceRight vals) prop = map (const $ Record.get prop vals)
+
+replaceRight :: forall rvals rin rout.
+  HMapWithIndex (ReplaceRight rvals) { | rin } { | rout } =>
+  { | rvals } ->
+  { | rin  } ->
+  { | rout }
+replaceRight =
+  hmapWithIndex <<< ReplaceRight
+
+testReplaceRight :: _
+testReplaceRight =
+  { a: "goodbye"
+  , b: 100
+  }
+  `replaceRight`
+  { a: Left "hello"
+  , b: Right 1
+  }
+
+testReplaceBoth :: forall rvals rin rout rs.
+  RowToList rout rs =>
+  HMapWithIndex (ReplaceLeft rvals) { | rin } { | rout } =>
+  HMapWithIndex (ReplaceRight rvals) { | rin } { | rout } =>
+  MapRecordWithIndex rs (ReplaceRight rvals) rout rout =>
+  { | rvals } ->
+  { | rin  } ->
+  { | rout }
+testReplaceBoth vals = replaceRight vals <<< replaceLeft vals
```

Comment:
I haven't been able to figure out why the `RowToList` and `MapRecordWithIndex` constraints are necessary. In addition, it's extremely brittle: `MapRecordWithIndex rs (ReplaceLeft rvals) rout rout` does not work, for example.